server/coordinator/procedure/create_table.go

```diff
@@ -42,63 +42,66 @@ var (
 
 func createTablePrepareCallback(event *fsm.Event) {
 	req := event.Args[0].(*createTableCallbackRequest)
-	table, exists, err := req.cluster.GetTable(req.ctx, req.sourceReq.GetSchemaName(), req.sourceReq.GetName())
+	_, exists, err := req.cluster.GetTable(req.sourceReq.GetSchemaName(), req.sourceReq.GetName())
 	if err != nil {
 		cancelEventWithLog(event, err, "cluster get table")
 		return
 	}
 	if exists {
-		req.createTableResult = &cluster.CreateTableResult{
-			Table: table,
-			ShardVersionUpdate: &cluster.ShardVersionUpdate{
-				ShardID: table.GetShardID(),
-			},
-		}
 		log.Warn("create an existing table", zap.String("schema", req.sourceReq.GetSchemaName()), zap.String("table", req.sourceReq.GetName()))
+		cancelEventWithLog(event, ErrTableAlreadyExists, "cluster get table")
 		return
 	}
 
 	createTableResult, err := req.cluster.CreateTable(req.ctx, req.sourceReq.GetHeader().GetNode(), req.sourceReq.GetSchemaName(), req.sourceReq.GetName())
 	if err != nil {
-		cancelEventWithLog(event, err, "cluster get table")
+		cancelEventWithLog(event, err, "cluster create table")
 		return
 	}
 
-	shard, err := req.cluster.GetShardByID(createTableResult.Table.GetShardID())
+	shardNodes, err := req.cluster.GetShardNodesByShardID(createTableResult.ShardVersionUpdate.ShardID)
 	if err != nil {
-		cancelEventWithLog(event, err, "cluster get shard by id")
+		cancelEventWithLog(event, err, "cluster get shardNode by id")
 		return
 	}
 	// TODO: consider followers
-	leader := shard.GetLeader()
-	if leader == nil {
-		cancelEventWithLog(event, ErrShardLeaderNotFound, "shard can't find leader")
+	leader := storage.ShardNode{}
+	found := false
+	for _, shardNode := range shardNodes {
+		if shardNode.ShardRole == storage.Leader {
```

Comment:
The meaning of this variable `storage.Leader`  is not clear, change it to `storage.ShardRole_Leader`